components/tidb_query_vec_aggr/src/impl_max_min.rs

```diff
@@ -281,6 +387,49 @@ mod tests {
         assert_eq!(result[0].as_int_slice(), &[Some(-1i64)]);
     }
 
+    #[test]
+    fn test_collation() {
+        let mut ctx = EvalContext::default();
+        let cases = vec![
+            (Collation::Binary, true, vec!["A", "a"], "a"),
+            (Collation::Utf8Mb4Bin, true, vec!["A", "a"], "a"),
+            (Collation::Utf8Mb4GeneralCi, true, vec!["A", "a"], "A"),
+            (Collation::Utf8Mb4BinNoPadding, true, vec!["A", "a"], "a"),
+            (Collation::Binary, false, vec!["a", "A"], "A"),
+            (Collation::Utf8Mb4Bin, false, vec!["a", "A"], "A"),
+            (Collation::Utf8Mb4GeneralCi, false, vec!["a", "A"], "a"),
+            (Collation::Utf8Mb4BinNoPadding, false, vec!["a", "A"], "A"),
+        ];
+        for (coll, is_max, args, expected) in cases {
+            let function = match_template_collator! {
+                TT, match coll {
+                    Collation::TT => {
+                        match is_max {
+                            true => Box::new(AggFnExtremum4Bytes::<TT, Max>::new()) as Box<dyn AggrFunction>,
+                            false => Box::new(AggFnExtremum4Bytes::<TT, Min>::new()) as Box<dyn AggrFunction>
```

Comment:
Silent `make clippy` seems not working well but `PROST=1 make clippy` complains that
```
[2020-06-11T12:09:11.818Z] 407 | /                         match is_max {
[2020-06-11T12:09:11.818Z] 408 | |                             true => Box::new(AggFnExtremum4Bytes::<TT, Max>::new()) as Box<dyn AggrFunction>,
[2020-06-11T12:09:11.818Z] 409 | |                             false => Box::new(AggFnExtremum4Bytes::<TT, Min>::new()) as Box<dyn AggrFunction>
[2020-06-11T12:09:11.818Z] 410 | |                         }
[2020-06-11T12:09:11.818Z]     | |_________________________^ help: consider using an `if`/`else` expression: `if is_max { Box::new(AggFnExtremum4Bytes::<TT, Max>::new()) as Box<dyn AggrFunction> } else { Box::new(AggFnExtremum4Bytes::<TT, Min>::new()) as Box<dyn AggrFunction> }`
```